packages/core/src/asset/AssetPromise.ts

```diff
@@ -2,20 +2,26 @@
  * Asset Loading Promise.
  */
 export class AssetPromise<T> implements PromiseLike<T> {
+  static resolve<T = any>(value?: T | PromiseLike<T>): AssetPromise<T> {
```

Comment:
Using function overloading and write comments

Please refer to Promise type redesign：
![image](https://github.com/user-attachments/assets/ea24812c-1bee-4f24-bc19-60c394e6dde6)